src/libraries/System.Text.RegularExpressions/tests/RegexReductionTests.cs

```diff
@@ -124,6 +124,16 @@ private static int GetMinRequiredLength(Regex r)
         [InlineData("aa+", "a{2,}")]
         [InlineData("aa?", "a{1,2}")]
         [InlineData("aa{1,3}", "a{2,4}")]
+        [InlineData("aaa+b", "aa(?>a+)b")]
```

Comment:
would this also reduce the same as `(?>aaa)a*b`

Reply:
No:
https://github.com/dotnet/runtime/pull/65273/files#diff-69d0a75aaf0674d30e057b6ac1d37203ee4757db089bac19f32a1d31d52a10c9R1689-R1692